testplan/web_ui/testing/src/state/appSlice.js

```diff
@@ -0,0 +1,85 @@
+// @ts-nocheck
+import { createSlice } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import { headers as DefaultHeaders } from 'axios/lib/defaults';
+import { flattened } from '../Common/utils';
+
+const __DEV__ = process.env.NODE_ENV !== 'production';
+const API_BASE_URL = process.env.REACT_APP_API_BASE_URL;
+
+/** always false in production */
+const boolIfDev = val => __DEV__ ? !!val : false;
+
+/**
+ * This state slice contains app-wide information that could be used in any
+ * part of the app
+ */
+export default (() => {
+  let apiBaseUrlObj = null;
+  if(typeof API_BASE_URL === 'string' && API_BASE_URL.length) {
```

Comment:
I see a lot of code to accommodate dev mode logic, can we reduce? Or at least make it very clear that it's just for dev mode purpuse. e.g env var for dev mode can start with DEV, dev mode code be embraced in a big if(DEV) block.

Reply:
moved dev stuff into a single block so easier to ignore if wanted